rewrite-gradle/src/main/java/org/openrewrite/gradle/AddGradleWrapper.java

```diff
@@ -88,6 +92,12 @@ public AddGradleWrapper(String version, String distribution) {
         this.distribution = distribution;
     }
 
+    @LoathingOfOthers("JavaDoc")
+    @JsonCreator
+    public AddGradleWrapper() {
```

Comment:
So the typical way of handling this would be to check if version is null or empty in the `getVisitor()` method. If null or empty, set version to RELEASE and pass into the same existing flow.

The `VersionComparator` class already has all of the necessary logic to select the latest release version, so we don't have to make a separate alternative HTTP invocation.

By doing this that means, that we don't need a separate constructor either.

Reply:
When I tried it, this block

```java

GradleVersion gradleVersion = allVersions.stream()
                            .filter(v -> versionComparator.isValid(null, v.version))
                            .max((v1, v2) -> versionComparator.compare(null, v1.version, v2.version))
                            .orElseThrow(() -> new IllegalStateException("Expected to find at least one Gradle wrapper version to select from."));

```

Failed to parse it as it did not find any matching release from `allVersions` 

``allVersions`` is an array of versions but I did not find any attribute stating that it would be the latest or not